devices/tuya.js

```diff
@@ -2477,41 +2477,71 @@ module.exports = [
+                .withLocalTemperatureCalibration(-30, 30, 0.1, ea.STATE_SET)
+                .withPreset(['auto', 'manual', 'off', 'on'],
                     'MANUAL MODE ☝ - In this mode, the device executes manual temperature setting. ' +
-                    'When the set temperature is lower than the "minimum temperature", the valve is closed (forced closed). ' +
-                    'AUTO MODE ⏱ - In this mode, the device executes a preset week programming temperature time and temperature. ' +
-                    'ON - In this mode, the thermostat stays open ' +
-                    'OFF - In this mode, the thermostat stays closed')
-                .withSystemMode(['auto', 'heat', 'off'], ea.STATE_SET),
-            exposes.composite('programming_mode', 'programming_mode', ea.STATE).withDescription('Auto MODE ⏱ - In this mode, ' +
-                    'the device executes a preset week programming temperature time and temperature. ')
-                .withFeature(exposes.text('monday_schedule', ea.STATE))
-                .withFeature(exposes.text('tuesday_schedule', ea.STATE))
-                .withFeature(exposes.text('wednesday_schedule', ea.STATE))
-                .withFeature(exposes.text('thursday_schedule', ea.STATE))
-                .withFeature(exposes.text('friday_schedule', ea.STATE))
-                .withFeature(exposes.text('saturday_schedule', ea.STATE))
-                .withFeature(exposes.text('sunday_schedule', ea.STATE)),
-            exposes.binary('boost_heating', ea.STATE_SET, 'ON', 'OFF').withDescription('Boost Heating: press and hold "+" for 3 seconds, ' +
+                'When the set temperature is lower than the "minimum temperature", the valve is closed (forced closed). ' +
+                'AUTO MODE ⏱ - In this mode, the device executes a preset week programming temperature time and temperature. ' +
+                'ON - In this mode, the thermostat stays open ' +
+                'OFF - In this mode, the thermostat stays closed')
+                .withSystemMode(['auto', 'heat', 'off'], ea.STATE)
+                .withRunningState(['idle', 'heat'], ea.STATE),
+            ...tuya.exposes.scheduleAllDays(ea.STATE_SET, 'HH:MM/C HH:MM/C HH:MM/C HH:MM/C'),
+            exposes.binary('boost_heating', ea.STATE_SET, 'ON', 'OFF')
+                .withDescription('Boost Heating: press and hold "+" for 3 seconds, ' +
                 'the device will enter the boost heating mode, and the ▷╵◁ will flash. The countdown will be displayed in the APP'),
-            exposes.numeric('boost_heating_countdown', ea.STATE_SET).withUnit('min').withDescription('Countdown in minutes')
+            exposes.numeric('boost_time', ea.STATE_SET).withUnit('min').withDescription('Countdown in minutes')
                 .withValueMin(0).withValueMax(1000),
         ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'system_mode',
+                    {
+                        from: (v) => {
```

Comment:
Can this be solved with the `tuya.valueConverterBasic.lookup`? (what is the problem when used)

Reply:
I do not think so. Because I will need create map with two `auto` keys. The problem is that the TRV only reports `preset` not the `system_mode` so I need to do some mapping. If I do not report `system_mode` HVAC modes in HA is not set correctly and show `cooling` etc. And `system_mode` does not allow `manual` value. Any ideas about that?

Reply:
You can also return the preset in this converter, to return multiple keys from `from` see e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/712ce49bff609b5b65516f5491f496993f95322a/lib/tuya.js#L1302

> And system_mode does not allow manual value. Any ideas about that?

How was this previously handled?

Reply:
It works same as currently. When reading, set both `system_mode` and `preset`. When writing `system_mode` or `preset` set same value.

Reply:
Yes you can, it will also change `preset` to `on`

Reply:
In HA I only change `preset`. Does not use HVAC mode (`system_mode`).

Reply:
I see, is this PR ready for merge now?

Reply:
Yes, it is.